### ISSUE TITLE
Facebook logout doesn't clear all the cookies

### ISSUE THREAD

**Issue by chiranshu14** [abacritt/angularx-social-login#397]
Hello Team,

I  neeed to implement an interface where a user can connect to multiple facebook accounts. With the default way of implementing facebook login, it asks for credentials only the first time (and it works fine). But when I click the same button to add my second account, it does not show a popup, but it directly logs in with the existing user.
So I tried multiple things to resolve this - 
1. Logout after signing in - 
After getting the access token for account 1, I store it in the backend and hence I can safely logout from angular. I call the logout function as suggested in the docs, it does the following -
- resets the session variable - removes the token from session variable
- modifies one of the cookies to Y
- does not clear all the existing cookies that were generated by facebook

So the logout function does work, but I still do not get option to enter credentials next time.

2. Logout before sign in - does the same steps as 1, but I still do not get the popup that would ask for credentials

3. auth_type: 'reauthenticate' - I tried passing in the sigin function, it does show me the popup window but it does not give option to switch to a different user.  
![image](https://user-images.githubusercontent.com/25884862/119244197-b404e500-bb8b-11eb-8639-f188a7cbd9d1.png)

(Also, I noticed that the UI of the login window looks different from what I see in production applications.)

4. Manually delete the "xs" cookie - 
manually deleting the "xs" cookie (or all of the cookies) works fine, after deleting it I always get the popup window.

So I'm pretty sure that the logout function should have deleted the session variables and the Facebook cookies, which it doesn't. 

**Question -** 
1. How can we delete these cookies from typescript, I read that we cannot delete HttpOnly cookies. Should be done from server side with logout. 
OR
2. Can we login without setting the cookies? (with traditional authentication, if we do not use withCredentials:true, we won't save the cookies. Can we do something like that?

Any help would be appreciated.

Thank you!
Chiranshu

**Comment by jaibatrik** [comment]
Hey @chiranshu14, I am not sure if this library will be helpful in your case. The library assumes there would be only one user at a time, and that's baked into the code.

As far as disconnecting a user goes, I think the Facebook provider `signOut` method can accept a boolean argument that does sort of a hard logout. Check if that helps you.

**Comment by chiranshu14** [comment]
Hi @jaibatrik 

Thanks for your quick response.

The login works fine, I just need to be able to disconnect from the user. 

I tried the provided signOut(true) function and it clears the auth_token in the session variable.
But it does not clear or delete the cookies. Unfortunately, I can't delete cookies from js/ts as it is an HttpOnly cookie. 
Ideally, the logout function should remove the cookies as well. (That's what happens in my traditional authentication modules)

Can you please suggest if there is any way to get rid of the cookies while logging out?

Thank you!

**Comment by jaibatrik** [comment]
Have been out of touch regarding the individual provider code for some time now. Not able to remember anything related to clearing cookies 😔

**Comment by chiranshu14** [comment]
okay, I'll try to build the login flow manually. 

Appreciate your time!
Thanks

**Comment by chiranshu14** [comment]
@jaibatrik 
Hi Jaibatrik!

For a general login scenario, if a person is already logged into one facebook account (acc1). But while logging into your application he wants to use a different account (acc2). The FB.login() takes the acc1 by default and just proceeds further automatically. How would you give the user an option to switch user?

**Comment by jaibatrik** [comment]
> @jaibatrik
> Hi Jaibatrik!
> 
> For a general login scenario, if a person is already logged into one facebook account (acc1). But while logging into your application he wants to use a different account (acc2). The FB.login() takes the acc1 by default and just proceeds further automatically. How would you give the user an option to switch user?

I haven't tried this, but vaguely remember that the boolean paramaeter to `signOut` was introduced to address this use case. The problem you are describing must be there with the Facebook JS SDK itself. Do you know how it addresses that? Need to figure it out and incorporate it here probably.

**Comment by chiranshu14** [comment]
To me, the boolean parameter didn't make any difference. signOut(true) will just clear the token from Session variables. It doesn't log the user out from facebook.com (in the browser scope). 
For us to be able to switch to another account we need to clear the cookies. Mainly the 'xs' cookie, manually deleting it automatically logs the user off from fb.com. We can't do that from js ourselves as it is HttpOnly cookie.

This is a very weird issue, fb docs say that signOut(true) will sign out globally, but doesn't work that way. And I've tried it from https (locally and stackblitz), so it's not about ssl, etc.

btw, I tried doing the same thing with the sdk directly and still didn't work.
